src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java

```diff
@@ -131,19 +138,29 @@ protected void initSearchController() {
     searchController.setVaultRoot(vaultRoot);
     searchController.setSavedQueries(vaultPrefs.getSavedReplayQueries());
 
-    searchController.addTextFilter("playerStats.player.login", i18n.get("game.player.username"), true);
-    searchController.addTextFilter("mapVersion.map.displayName", i18n.get("game.map.displayName"), false);
-    searchController.addTextFilter("mapVersion.map.author.login", i18n.get("game.map.author"), false);
-    searchController.addTextFilter("name", i18n.get("game.title"), false);
-    searchController.addTextFilter("id", i18n.get("game.id"), true);
+    ReplaySearchPrefs replaySearchPrefs = vaultPrefs.getReplaySearch();
+    TextFilterController textFilterController = searchController.addTextFilter("playerStats.player.login", i18n.get("game.player.username"), true);
+    textFilterController.textFieldProperty().bindBidirectional(replaySearchPrefs.playerNameFieldProperty());
+    textFilterController = searchController.addTextFilter("mapVersion.map.displayName", i18n.get("game.map.displayName"), false);
+    textFilterController.textFieldProperty().bindBidirectional(replaySearchPrefs.mapNameFieldProperty());
+    textFilterController = searchController.addTextFilter("mapVersion.map.author.login", i18n.get("game.map.author"), false);
+    textFilterController.textFieldProperty().bindBidirectional(replaySearchPrefs.mapAuthorFieldProperty());
+    textFilterController = searchController.addTextFilter("name", i18n.get("game.title"), false);
+    textFilterController.textFieldProperty().bindBidirectional(replaySearchPrefs.titleFieldProperty());
+    textFilterController = searchController.addTextFilter("id", i18n.get("game.id"), true);
+    textFilterController.textFieldProperty().bindBidirectional(replaySearchPrefs.replayIDFieldProperty());
 
     CategoryFilterController featuredModFilterController = searchController.addCategoryFilter("featuredMod.displayName",
         i18n.get("featuredMod.displayName"), List.of());
 
     featuredModService.getFeaturedMods().map(FeaturedMod::displayName)
                       .collectList()
                       .publishOn(fxApplicationThreadExecutor.asScheduler())
-                      .subscribe(featuredModFilterController::setItems);
+                      .subscribe((mods) -> {
+                        featuredModFilterController.setItems(mods);
+                        replaySearchPrefs.featuredModFilterProperty().get().stream().forEach((item) -> featuredModFilterController.checkItem(item));
+                        replaySearchPrefs.featuredModFilterProperty().bind(Bindings.createObjectBinding(() -> featuredModFilterController.getCheckedItems()));
```

Comment:
There is a utility method Bindings.bindContent that can be used here. Although we have to be more careful about unbinding when the vault controller is closed. This could be easily done by adding the unsubscription with the addSubscription method of the controller

Reply:
I dont understand whats needed about unbinding

Reply:
ahh, to prevent memory leaks as above

Reply:
I don't understand how to add an unsubscription